sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -192,19 +192,96 @@ def test_baseline_categorical_crossentropy():
     rng = np.random.RandomState(0)
 
     prediction_dim = 4
-    loss = _LOSSES['categorical_crossentropy']()
+    loss = _LOSSES['categorical_crossentropy'](sample_weight=None)
     for y_train in (np.zeros(shape=100), np.ones(shape=100)):
         y_train = y_train.astype(np.float64)
-        baseline_prediction = loss.get_baseline_prediction(y_train,
+        baseline_prediction = loss.get_baseline_prediction(y_train, None,
                                                            prediction_dim)
         assert baseline_prediction.dtype == y_train.dtype
         assert_all_finite(baseline_prediction)
 
     # Same logic as for above test. Here inverse_link_function = softmax and
     # link_function = log
     y_train = rng.randint(0, prediction_dim + 1, size=100).astype(np.float32)
-    baseline_prediction = loss.get_baseline_prediction(y_train, prediction_dim)
+    baseline_prediction = loss.get_baseline_prediction(y_train, None,
+                                                       prediction_dim)
     assert baseline_prediction.shape == (prediction_dim, 1)
     for k in range(prediction_dim):
         p = (y_train == k).mean()
         assert np.allclose(baseline_prediction[k, :], np.log(p))
+
+
+@pytest.mark.parametrize('loss, problem', [
+    ('least_squares', 'regression'),
+    ('least_absolute_deviation', 'regression'),
+    ('binary_crossentropy', 'classification'),
+    ('categorical_crossentropy', 'classification')
+    ])
+@pytest.mark.parametrize('sample_weight', ['ones', 'random'])
+def test_sample_weight_multiplies_gradients(loss, problem, sample_weight):
+    # Make sure that passing sample weights to the gradient and hessians
+    # computation methods is equivalent to multiplying by the weights.
+
+    rng = np.random.RandomState(42)
+    n_samples = 1000
+
+    if loss == 'categorical_crossentropy':
+        n_classes = prediction_dim = 3
+    else:
+        n_classes = prediction_dim = 1
+
+    if problem == 'regression':
+        y_true = rng.normal(size=n_samples).astype(Y_DTYPE)
+    else:
+        y_true = rng.randint(0, n_classes, size=n_samples).astype(Y_DTYPE)
+
+    if sample_weight == 'ones':
+        sample_weight = np.ones(shape=n_samples, dtype=Y_DTYPE)
+    else:
+        sample_weight = rng.normal(size=n_samples).astype(Y_DTYPE)
```

Comment:
Do we want to include negative numbers here?

Reply:
There doesn't seem to be a place where we check for non-negative weights, and I'm not sure if the math requires the weights to be positive.

Reply:
For ref, this is discussed in https://github.com/scikit-learn/scikit-learn/issues/15531

I think we can safely ignore the topic for now. (I'm reasonably confident that negative SW would work "as expected" here, though I've no idea what it would look like with binning)